### ISSUE TITLE
More conscious scaffolding of concepts

### ISSUE THREAD

**Issue by benjimelito** [ga-wdi-boston/js-array-methods#40]
I noticed that a large number of developers felt lost on the diagnostic for this topic, and many of them requested 1:1s to go over it. A lot of the feedback that I received seemed to suggest that they were confused by the introduction of callbacks, anonymous functions, array iteration methods, and arrow functions all at once. I feel that it might be beneficial to either introduce some of these concepts earlier, or to more consciously/slowly scaffold them in the delivery itself.

**Comment by payne-chris-r** [comment]
I think the main problem is much of this is in @gaand 's head, and it needs to not be. We need to solve this before next iteration.

**Comment by payne-chris-r** [comment]
The more I think about it, the more I think there should be ~4 sections to the iterations methods talk.
1) what are some examples
2) what is that thing we're passing to them? (`a callback`)
3) what is a fat arrow function, and can we also pass those to functions? (yes)
4) how do these functions actually work?

**Comment by gaand** [comment]
No. This should be covered before array  iteration methods. I thought it was in function ins and outs (accept arrow functions).

**Comment by payne-chris-r** [comment]
Aye. That is true. Maybe just 1 and 4, then?

**Comment by gaand** [comment]
Please consider my comment in slack:  

> If you're going to suggest adding to material we currently present, please tell me what you intend to remove to make room.

This is not push back.

**Comment by benjimelito** [comment]
From my very limited experience, it seems like comparing an anonymous function passed as a callback (to an array iteration method) to a named function passed as a callback was helpful to the developers, as was comparing an arrow function to a standard function. 

With respect to what @gaand said regarding taking something out in order to include these additions, I don't have anything specific, but I would say that in general I feel it's more important that they understand the concept of an array iteration method than that they know a large number of these methods. In other words, maybe cover fewer methods?

(edit) Though I suppose 'comparing an anonymous function passed as a callback to a named function passed as a callback' would fall more under the scope of functions ins and outs.

**Comment by payne-chris-r** [comment]
Yeah Ben, I think that's what @gaand is suggesting. I don't think anyone is questioning that this could've been better covered, we're just trying to figure out where it belongs, and where the gaps are. This type of thing _should_ be covered in js-functions-arrays-ins-and-outs. I've added it there and linking here: https://github.com/ga-wdi-boston/js-functions-ins-and-outs/issues/37
